src/caldav-calendar.c

```diff
@@ -396,7 +396,7 @@ static Event* create_event_from_parsed_xml(CaldavCalendar* cal, CaldavEntry* cde
 	// event updated
 	icalcomponent* comp = icalparser_parse_string(cde->caldata);
 	icalcomponent* vev = icalcomponent_get_first_component(comp, ICAL_VEVENT_COMPONENT);
-	g_assert_nonnull(vev);
+	if (!vev) return NULL;
```

Comment:
nitpick: focal [code style](https://github.com/ohwgiles/focal/blob/master/CONTRIBUTING.md#code-style) would put this on a second line. Good idea to run `clang-format` before pushing